JetNews/app/src/main/java/com/example/jetnews/ui/SwipeToRefresh.kt

```diff
@@ -16,212 +16,77 @@
 
-import androidx.animation.AnimatedFloat
-import androidx.animation.AnimationClockObservable
-import androidx.animation.AnimationEndReason
-import androidx.animation.AnimationSpec
-import androidx.animation.ExponentialDecay
-import androidx.animation.OnAnimationEnd
-import androidx.animation.Spring
-import androidx.animation.TargetAnimation
-import androidx.animation.TweenSpec
-import androidx.annotation.FloatRange
-import androidx.compose.Composable
-import androidx.compose.onCommit
-import androidx.compose.remember
-import androidx.compose.state
-import androidx.ui.animation.asDisposableClock
-import androidx.ui.core.Alignment
-import androidx.ui.core.AnimationClockAmbient
-import androidx.ui.core.DensityAmbient
-import androidx.ui.core.Modifier
-import androidx.ui.core.composed
-import androidx.ui.core.gesture.scrollorientationlocking.Orientation
-import androidx.ui.foundation.Box
-import androidx.ui.foundation.InteractionState
-import androidx.ui.foundation.animation.FlingConfig
-import androidx.ui.foundation.animation.fling
-import androidx.ui.foundation.gestures.draggable
-import androidx.ui.layout.Stack
-import androidx.ui.layout.offsetPx
-import androidx.ui.unit.dp
-import androidx.ui.util.fastFirstOrNull
-import androidx.ui.util.lerp
+import androidx.compose.animation.asDisposableClock
+import androidx.compose.animation.core.AnimationClockObservable
+import androidx.compose.foundation.Box
+import androidx.compose.foundation.layout.Stack
+import androidx.compose.foundation.layout.offsetPx
+import androidx.compose.material.ExperimentalMaterialApi
+import androidx.compose.material.FractionalThreshold
+import androidx.compose.material.SwipeableState
+import androidx.compose.material.swipeable
+import androidx.compose.runtime.Composable
+import androidx.compose.runtime.onCommit
+import androidx.compose.runtime.remember
+import androidx.compose.ui.Alignment
+import androidx.compose.ui.Modifier
+import androidx.compose.ui.gesture.scrollorientationlocking.Orientation
+import androidx.compose.ui.platform.AnimationClockAmbient
+import androidx.compose.ui.platform.DensityAmbient
+import androidx.compose.ui.unit.dp
 
-private val SWIPE_DISTANCE_SIZE = 100.dp
-private const val SWIPE_DOWN_OFFSET = 1.2f
+private val RefreshDistance = 80.dp
 
+@OptIn(ExperimentalMaterialApi::class)
 @Composable
 fun SwipeToRefreshLayout(
     refreshingState: Boolean,
     onRefresh: () -> Unit,
-    refreshIndicator: @Composable() () -> Unit,
-    dragCoefficient: Float = 0.5f,
-    content: @Composable() () -> Unit
+    refreshIndicator: @Composable () -> Unit,
+    content: @Composable () -> Unit
 ) {
-    val size = with(DensityAmbient.current) { SWIPE_DISTANCE_SIZE.toPx() }
-    // min is below negative to hide
-    val min = -size
-    val max = size * SWIPE_DOWN_OFFSET
-    val dragPosition = state { max }
+    val refreshDistance = with(DensityAmbient.current) { RefreshDistance.toPx() }
+    val state = rememberRefreshSwipeableState(refreshingState)
```

Comment:
The issue is that this needs to update if the `refreshingState` argument changes which is why I didn't use `rememberSwipeableState` so that I could guard it's `remember` with that arg.

Reply:
Ah! I see, makes sense. Let's leave it like this until we have a better alternative

Reply:
https://issuetracker.google.com/issues/164113833